wdl-lint/src/rules/nonmatching_output.rs

```diff
@@ -0,0 +1,349 @@
+
+        // Reset the visitor upon document entry
+        *self = Default::default();
+    }
+
+    fn workflow_definition(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        workflow: &WorkflowDefinition,
+    ) {
+        match reason {
+            VisitReason::Enter => {
+                self.name = Some(workflow.name().as_str().to_string());
+                self.ty = Some("workflow");
+            }
+            VisitReason::Exit => {
+                if self.current_meta_span.is_some()
+                    && self.current_meta_outputs_span.is_none()
+                    && !self.output_keys.is_empty()
+                {
+                    state.add(missing_outputs_in_meta(
+                        self.current_meta_span.expect("should have a `meta` span"),
+                        self.name.as_deref().expect("should have a name"),
+                        self.ty.expect("should have a type"),
+                    ));
+                } else {
+                    check_matching(state, self);
+                }
+
+                self.name = None;
+                self.current_meta_outputs_span = None;
+                self.current_meta_span = None;
+                self.current_output_span = None;
+                self.output_keys.clear();
+                self.meta_outputs_keys.clear();
+            }
+        }
+    }
+
+    fn task_definition(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        task: &TaskDefinition,
+    ) {
+        match reason {
+            VisitReason::Enter => {
+                self.name = Some(task.name().as_str().to_string());
+                self.ty = Some("task");
+            }
+            VisitReason::Exit => {
+                if self.current_meta_span.is_some()
+                    && self.current_meta_outputs_span.is_none()
+                    && !self.output_keys.is_empty()
+                {
+                    state.add(missing_outputs_in_meta(
+                        self.current_meta_span.expect("should have a `meta` span"),
+                        self.name.as_deref().expect("should have a name"),
+                        self.ty.expect("should have a type"),
+                    ));
+                } else {
+                    check_matching(state, self);
+                }
+
+                self.current_meta_outputs_span = None;
+                self.current_meta_span = None;
+                self.current_output_span = None;
+                self.output_keys.clear();
+                self.meta_outputs_keys.clear();
+            }
+        }
+    }
+
+    fn metadata_section(
+        &mut self,
+        _state: &mut Self::State,
+        reason: VisitReason,
+        section: &MetadataSection,
+    ) {
+        match reason {
+            VisitReason::Enter => {
+                self.current_meta_span = Some(section.syntax().text_range().to_span());
+            }
+            VisitReason::Exit => {}
```

Comment:
Just to be explicit and avoid future issues, would this be a place to set `self.current_meta_span = None`? It's probably inconsequential unless something is terribly wrong, but it's worth considering being explicit.

If the answer above is "no", then I feel this should at least be an if statement like in `output_section`.

Reply:
If so, it would require a refactor of the `Exit` logic for both `Task` and `Workflow`.